### DISCUSSION TITLE
Libcurl and curl binaries generates malformed packages on ubuntu 24.04 over httpv6

### DISCUSSION THREAD

**Discussion by mohneesh-9797** [curl/curl#15678] [Q&A]
I am trying to get the binaries from the v6 address via http protocol using the curl binaries and libcurl libraries but it seems some malformed packages are formed and the download fails after approx. 60% of data is transferred via libcurl.

I am using the **curl-8.9.1** and **curl -O -g -6 http://[2023:f:b:2267:ad79:1536:e6e0:b2a]/w/MINET_6940w-enc.st**  command to download my image.

In case of libcurl I have used **curl_easy_setopt()** function to set **CURLOPT_SHARE, CURLOPT_URL, CURLOPT_IPRESOLVE(to CURL_IPRESOLVE_V6), CURLOPT_TFTP_BLKSIZE, CURLOPT_NOSIGNAL, CURLOPT_NOSIGNAL, CURLOPT_TIMEOUT, CURLOPT_CONNECTTIMEOUT_MS, CURLOPT_HEADERDATA, CURLOPT_HEADERFUNCTION, CURLOPT_PROXY, CURLOPT_PROXYAUTH, CURLOPT_HTTPHEADER, CURLOPT_PASSWORD, CURLOPT_USERNAME, CURLOPT_UNRESTRICTED_AUTH, CURLOPT_FOLLOWLOCATION, CURLOPT_FORBID_REUSE, CURLOPT_DEBUGFUNCTION, CURLOPT_VERBOSE, CURLOPT_USERAGENT, CURLOPT_HEADER, CURLOPT_HTTPGET**  

I am using **curl_multi_add_handle()** function for handling multiple calls and **curl_easy_getinfo()** function to get the binary, I have called **curl_easy_getinfo()** function multiple times until I completely downloads my binary 

The screenshots for wireshark output : 
**Malformed packages:** 

![image](https://github.com/user-attachments/assets/fe8ebae6-0a02-45fa-bbd8-4acdad3426f9)
![image](https://github.com/user-attachments/assets/a06206ae-0b68-4117-8833-796c539e60d6)

Is there any parameter of function that we need to run in case of httpv6 so that malformed packages will not formed or is there  any other way to resolve this issue

**Comment by icing** [comment]
I know of no way how curl could trigger a malformed TCP packets. This points, as far as I can tell, more to a problem in your network configuration.
